extension/src/server/GradleServer.ts

```diff
@@ -29,25 +31,38 @@ export class GradleServer {
     constructor(
         private readonly opts: ServerOptions,
         private readonly context: vscode.ExtensionContext,
-        private readonly logger: Logger
+        private readonly logger: Logger,
+        private messageProxy: MessageProxy
     ) {}
 
     public async start(): Promise<void> {
         if (isDebuggingServer()) {
-            this.port = 8887;
             this.fireOnStart();
         } else {
-            this.port = await getPort();
+            this.gradleServerPort = await getPort();
             const cwd = this.context.asAbsolutePath("lib");
             const cmd = path.join(cwd, getGradleServerCommand());
             const env = await getGradleServerEnv();
+            const bundleDirectory = this.context.asAbsolutePath("server");
             if (!env) {
                 await vscode.window.showErrorMessage(NO_JAVA_EXECUTABLE);
                 return;
             }
-            const args = [String(this.port)];
-
-            this.logger.debug("Starting server");
+            let javaExecPath: string | null = null;
+            //Get the Java executable, which will be higher than JDK 17.
+            if (redHatJavaInstalled()) {
+                javaExecPath = getRedHatJavaExecutablePath() || (await getJavaExecutablePath());
+                if (javaExecPath === null) {
+                    await vscode.window.showErrorMessage(NO_JAVA_EXECUTABLE);
+                }
+            }
```

Comment:
logic here: 
if redHat.java installed: 
- Find embeded jre --(if not found)--> try other way to find java executable with `jdk>=17`
- if still not found, **showErrorMessage**

If not installed:
- meaning that `build server`&`importer` won't be invoked, but we still need to move forward and start MergeServer

Merge Server will check if java executable exist and decide whether start `build server` or not.